extensions/adapters/vector/src/main/java/mil/nga/giat/geowave/adapter/vector/AvroFeatureDataAdapter.java

```diff
@@ -0,0 +1,123 @@
+
+import java.util.ArrayList;
+import java.util.List;
+
+import mil.nga.giat.geowave.adapter.vector.plugin.visibility.JsonDefinitionColumnVisibilityManagement;
+import mil.nga.giat.geowave.adapter.vector.plugin.visibility.AvroAdaptorProxyFieldLevelVisibilityHandler;
+import mil.nga.giat.geowave.adapter.vector.utils.TimeDescriptors;
+import mil.nga.giat.geowave.core.geotime.store.dimension.Time;
+import mil.nga.giat.geowave.core.index.ByteArrayId;
+import mil.nga.giat.geowave.core.store.adapter.IndexFieldHandler;
+import mil.nga.giat.geowave.core.store.adapter.NativeFieldHandler;
+import mil.nga.giat.geowave.core.store.adapter.NativeFieldHandler.RowBuilder;
+import mil.nga.giat.geowave.core.store.adapter.PersistentIndexFieldHandler;
+import mil.nga.giat.geowave.core.store.data.field.FieldReader;
+import mil.nga.giat.geowave.core.store.data.field.FieldVisibilityHandler;
+import mil.nga.giat.geowave.core.store.data.field.FieldWriter;
+import mil.nga.giat.geowave.core.store.data.visibility.VisibilityManagement;
+import mil.nga.giat.geowave.core.store.index.CommonIndexValue;
+
+import org.opengis.feature.simple.SimpleFeature;
+import org.opengis.feature.simple.SimpleFeatureType;
+
+/**
+ */
+public class AvroFeatureDataAdapter extends
+		FeatureDataAdapter
+{
+
+	protected AvroFeatureDataAdapter() {}
+
+	public AvroFeatureDataAdapter(
+			final SimpleFeatureType type ) {
+		super(
+				type,
+				new ArrayList<PersistentIndexFieldHandler<SimpleFeature, ? extends CommonIndexValue, Object>>());
+	}
+
+	public AvroFeatureDataAdapter(
+			final SimpleFeatureType type,
+			final VisibilityManagement<SimpleFeature> visibilityManagement ) {
+		super(
+				type,
+				new ArrayList<PersistentIndexFieldHandler<SimpleFeature, ? extends CommonIndexValue, Object>>(),
+				null,
+				visibilityManagement);
+	}
+
+	public AvroFeatureDataAdapter(
+			final SimpleFeatureType type,
+			final List<PersistentIndexFieldHandler<SimpleFeature, ? extends CommonIndexValue, Object>> customIndexHandlers ) {
+		super(
+				type,
+				customIndexHandlers,
+				null,
+				new JsonDefinitionColumnVisibilityManagement<SimpleFeature>());
+	}
+
+	public AvroFeatureDataAdapter(
+			final SimpleFeatureType type,
+			final FieldVisibilityHandler<SimpleFeature, Object> fieldVisiblityHandler ) {
+		super(
+				type,
+				new ArrayList<PersistentIndexFieldHandler<SimpleFeature, ? extends CommonIndexValue, Object>>(),
+				fieldVisiblityHandler,
+				new JsonDefinitionColumnVisibilityManagement<SimpleFeature>());
+	}
+
+	@Override
+	protected List<NativeFieldHandler<SimpleFeature, Object>> typeToFieldHandlers(
+			final SimpleFeatureType type ) {
+		final List<NativeFieldHandler<SimpleFeature, Object>> nativeHandlers = new ArrayList<NativeFieldHandler<SimpleFeature, Object>>(
+				1);
+
+		nativeHandlers.add(new AvroFeatureAttributeHandler());
+		return nativeHandlers;
+	}
+
+	@Override
+	protected IndexFieldHandler<SimpleFeature, Time, Object> getTimeRangeHandler(
```

Comment:
Do we need to overload this method?  I don't think we need the AvroAdaptorProxyFieldLevelVisibilityHandler either.  The only difference between AvroAdaptorProxyFieldLevelVisibilityHandler and AdaptorProxyFieldLevelVisibilityHandler is the type of the data adapter.  However, I don't think we need to be specific about the adapter type in the Avro visibility handler because the AvroFeatureDataAdapter is a FeatureDataAdapter.  In short, I think we can reuse the AdaptorProxyFieldLevelVisibilityHandler in this method, but if we do that, the method remains unchanged from the base implementation, so I don't think we need to overload this method at all.  Hopefully that makes sense.  If not, let me know.